### ISSUE TITLE
Don't analyze hidden columns

### ISSUE THREAD

**Issue by franciscolourenco** [metabase/metabase#11903]
**Is your feature request related to a problem? Please describe.**
I have columns which have large amounts of data, but which are not interesting for querying. I would like to hide them, but metabase still tries to analyze them, and runs out of memory.
```
Error reading :postgres column OTHER
java.lang.OutOfMemoryError: Java heap space
```

**Describe the solution you'd like**
Don't analyze hidden columns

**Describe alternatives you've considered**
Disabling deep analysis globally, but would like to avoid doing that.

**How important is this feature to you?**
Quite, since metabase stops responding when this happens

**Additional context**
Related:
- https://github.com/metabase/metabase/issues/2127
- https://github.com/metabase/metabase/pull/4918

**Comment by sbelak** [comment]
Could you share the stacktrace as well, ideally we'd like to fix the underlying issue as well. 

#11832 should reduce memory consumption while fetching results.

**Comment by franciscolourenco** [comment]
@sbelak this or are you looking for something else?
```
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 INFO sync.util :: STARTING: Sync metadata for postgres Database 3 'Example'
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 INFO sync.util :: STARTING: step 'sync-timezone' for postgres Database 3 'Example'
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 INFO sync.util :: FINISHED: step 'sync-timezone' for postgres Database 3 'Example' (10
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 INFO sync.util :: STARTING: step 'sync-tables' for postgres Database 3 'Example'
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 INFO sync.util :: FINISHED: step 'sync-tables' for postgres Database 3 'Example' (39.0
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 INFO sync.util :: STARTING: step 'sync-fields' for postgres Database 3 'Example'
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."yes_
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."card
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:00 metabase metabase.sh[1610]: 02-12 20:50:00 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:01 metabase metabase.sh[1610]: 02-12 20:50:01 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:01 metabase metabase.sh[1610]: 02-12 20:50:01 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:01 metabase metabase.sh[1610]: 02-12 20:50:01 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:01 metabase metabase.sh[1610]: 02-12 20:50:01 WARN sql-jdbc.sync :: Don't know how to map column type '"information_schema"."sql_
Feb 12 20:50:01 metabase metabase.sh[1610]: 02-12 20:50:01 INFO sync.util :: FINISHED: step 'sync-fields' for postgres Database 3 'Example' (1.5
Feb 12 20:50:01 metabase metabase.sh[1610]: 02-12 20:50:01 INFO sync.util :: STARTING: step 'sync-fks' for postgres Database 3 'Example'
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.util :: FINISHED: step 'sync-fks' for postgres Database 3 'Example' (456.6 m
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.util :: STARTING: step 'sync-metabase-metadata' for postgres Database 3 '
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.util :: FINISHED: step 'sync-metabase-metadata' for postgres Database 3 '
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.util :: FINISHED: Sync metadata for postgres Database 3 'Example' (2.1 s)
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.util :: STARTING: Analyze data for postgres Database 3 'Example'
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.util :: STARTING: step 'fingerprint-fields' for postgres Database 3 'Bird
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.analyze :: fingerprint-fields Analyzed [*****····························
Feb 12 20:50:02 metabase metabase.sh[1610]: 02-12 20:50:02 INFO sync.analyze :: fingerprint-fields Analyzed [**********·······················
Feb 12 20:50:35 metabase metabase.sh[1610]: 02-12 20:50:35 ERROR sql-jdbc.execute :: Error reading :postgres column 4 column-a OTHER
Feb 12 20:50:35 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:36 metabase metabase.sh[1610]: 02-12 20:50:36 ERROR sql-jdbc.execute :: Error reading :postgres column 4 column-a OTHER
Feb 12 20:50:36 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:37 metabase metabase.sh[1610]: 02-12 20:50:37 ERROR sql-jdbc.execute :: Error reading :postgres column 19 column-b OTHER
Feb 12 20:50:37 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:39 metabase metabase.sh[1610]: 02-12 20:50:39 ERROR sql-jdbc.execute :: Error reading :postgres column 24 column-c VARCHAR
Feb 12 20:50:39 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:39 metabase metabase.sh[1610]: 02-12 20:50:39 ERROR sql-jdbc.execute :: Error reading :postgres column 4 column-a OTHER
Feb 12 20:50:39 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:41 metabase metabase.sh[1610]: 02-12 20:50:41 ERROR sql-jdbc.execute :: Error reading :postgres column 19 column-b OTHER
Feb 12 20:50:41 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:43 metabase metabase.sh[1610]: 02-12 20:50:43 ERROR sql-jdbc.execute :: Error reading :postgres column 6 column-d
Feb 12 20:50:43 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:44 metabase metabase.sh[1610]: 02-12 20:50:44 ERROR sql-jdbc.execute :: Error reading :postgres column 4 column-a OTHER
Feb 12 20:50:44 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:44 metabase metabase.sh[1610]: 02-12 20:50:44 WARN server.AbstractConnector ::
Feb 12 20:50:44 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:44 metabase metabase.sh[1610]: 02-12 20:50:44 ERROR sql-jdbc.execute :: Error reading :postgres column 19 column-b OTHER
Feb 12 20:50:44 metabase metabase.sh[1610]: java.lang.OutOfMemoryError: Java heap space
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.analyze :: fingerprint-fields Analyzed [****************·················
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.util :: FINISHED: step 'fingerprint-fields' for postgres Database 3 'Bird
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.util :: STARTING: step 'classify-fields' for postgres Database 3 'Example'
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.analyze :: classify-fields Analyzed [*********************···············
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.analyze :: classify-fields Analyzed [***************************·········
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.analyze :: classify-fields Analyzed [********************************····
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.util :: FINISHED: step 'classify-fields' for postgres Database 3 'Example' (
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.util :: STARTING: step 'classify-tables' for postgres Database 3 'Example'
Feb 12 20:50:46 metabase metabase.sh[1610]: 02-12 20:50:46 INFO sync.analyze :: classify-tables Analyzed [************************************
```

**Comment by franciscolourenco** [comment]
Reducing memory consumption would be a good thing, but is there a point in scanning hidden columns?

**Comment by sbelak** [comment]
That's definitely a bug. We have some logic that should prevent hidden fields being analyzed. I'll look into why this is not the case and get back to you if I need any more information.

**Comment by robdaemon** [comment]
This was fixed in our 0.34.0 release. Please try against a newer version.

The change was here: https://github.com/metabase/metabase/commit/bf3b25a6b7a2e1460f895033b900f0292310fc9d where we started excluding hidden fields (known as `sensitive` in our underlying database).

**Comment by robdaemon** [comment]
Was fixed as part of #10776
